### ISSUE TITLE
citrix_adc_service slow in large scale

### ISSUE THREAD

**Issue by vkovacik** [citrix/citrix-adc-ansible-modules#122]
**Issue:**
`citrix_adc_service` module takes long time to execute on netscaler with high number of configured services.

Test scenarios with:
- LB with 1043 services: `citrix_adc_service` took ~19 seconds
- LB with 224 services: `citrix_adc_service` took ~3 seconds

Apparently the module execution time is linearly dependent on number of services configured on netscaler. The above tests were running on a service which was already present on netscaler. There were no changes to be applied on this service.

Time profiling on LB with 1043 services:
```loglines:
  - Applying actions for state present
  - 'time(service_exists): 4196, result: True'
  - 'time(all_identical): 4812, result: True'
  - Enabling service
  - 'time(do_state_change): 146'
  - Sanity checks for state present
  - 'time(service_exists): 4304'
  - 'time(service_identical): 4947'
  - 'time(monitor_bindings_identical): 308'
  - 'time(all_nitro_apis): 19195'
```

Out of total 19195 ms, most of the time was spent in function calls:
- service_exists (4196+4947ms)
- all_indentical (4812ms)
- service_identical (4947ms)

These functions call the following time expensive Nitro APIs:
- service.count_filtered
- service_lbmonitor_binding.count

Although these API run on the backend, their execution time is linear based on the number of services in netscaler config.

**Proposal:**
Avoid running count APIs and rather fetch service object directly by name.

**Comment by vkovacik** [comment]
I already have proposed changes ready. I'll prepare PR.

**Comment by vkovacik** [comment]
@giorgos-nikolopoulos Can you review https://github.com/citrix/citrix-adc-ansible-modules/pull/123 ?

Before:
```
# ansible-playbook library/tests/test_citrix_adc_service.yml -v
...
TASK [Ensure service is present] ***********************************************************
Friday 05 June 2020  03:02:07 +0000 (0:00:00.064)       0:00:00.418 *********** 
ok: [localhost] => changed=false 
  loglines:
  - Applying actions for state present
  - Enabling service
  - Sanity checks for state present

PLAY RECAP *********************************************************************************
localhost                  : ok=3    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   

Friday 05 June 2020  03:02:23 +0000 (0:00:16.663)       0:00:17.081 *********** 
=============================================================================== 
Ensure service is present ---------------------------------------------------------- 16.66s
Lookup Netscaler primary node ------------------------------------------------------- 0.10s
Set service name -------------------------------------------------------------------- 0.06s
Playbook run took 0 days, 0 hours, 0 minutes, 17 seconds
```

After:
```
# ansible-playbook library/tests/test_citrix_adc_service.yml -v
...
TASK [Lookup Netscaler primary node] *******************************************************
Friday 05 June 2020  03:03:03 +0000 (0:00:00.260)       0:00:00.260 *********** 
ok: [localhost] => changed=false 
  ansible_facts:
    nsip: las1-lb-c1

TASK [Set service name] ********************************************************************
Friday 05 June 2020  03:03:03 +0000 (0:00:00.091)       0:00:00.352 *********** 
ok: [localhost] => changed=false 
  ansible_facts:
    service_name: REDACTED

TASK [Ensure service is present] ***********************************************************
Friday 05 June 2020  03:03:03 +0000 (0:00:00.064)       0:00:00.416 *********** 
ok: [localhost] => changed=false 
  loglines:
  - Applying actions for state present
  - Enabling service
  - Sanity checks for state present

PLAY RECAP *********************************************************************************
localhost                  : ok=3    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   

Friday 05 June 2020  03:03:05 +0000 (0:00:02.048)       0:00:02.464 *********** 
=============================================================================== 
Ensure service is present ----------------------------------------------------------- 2.05s
Lookup Netscaler primary node ------------------------------------------------------- 0.09s
Set service name -------------------------------------------------------------------- 0.06s
Playbook run took 0 days, 0 hours, 0 minutes, 2 seconds
```

**Comment by giorgos-nikolopoulos** [comment]
We are currently not accepting any third party contributions.

The `citrix_adc_service` module will have to be rewritten to improve efficiency.
We will take this up when time permits.

In the mean time you can use `citrix_adc_nitro_resource` or `citrix_adc_nitro_request` to configure services without the time overhead.

**Comment by giorgos-nikolopoulos** [comment]
Fixed with #178.

The `citrix_adc_service` module was rewritten.
There is no use now of `count_filtered`, which slowed down the processing.
